ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -795,17 +795,28 @@ abstract class EventStorageBackendTemplate(
   }
 
   override def maxEventSequentialIdOfAnObservableEvent(
-      offset: Offset
-  )(connection: Connection): Option[Long] = {
+      offsetInclusive: Offset
+  )(connection: Connection): Long = {
+    val ledgerEnd = ledgerEndCache()
     import com.daml.platform.store.backend.Conversions.OffsetToStatement
     SQL"""
-         SELECT
-            event_sequential_id_last
-         FROM
-            participant_transaction_meta
-         WHERE
-            event_offset = (SELECT MAX(event_offset) FROM participant_transaction_meta WHERE event_offset <= $offset)
-       """.as(get[Long](1).singleOpt)(connection)
+     SELECT
+        event_sequential_id_first
+     FROM
+        participant_transaction_meta
+     WHERE
+        event_offset > $offsetInclusive
```

Comment:
Not sure if I understand if why this is called `offsetInclusive`. This method sometimes gets passed `startExclusive` and sometimes `endInclusive`

Reply:
this is about the maxEventSequentialIdOfAnObservableEvent, which's offset parameter means before or equal, not only before.
But I agree we could make this more clear, how about untilInclusiveOffset?